crates/ext-processor/src/service.rs

```diff
@@ -299,6 +337,35 @@
         Err(PrepareResponseError::MissingHeaders)
     }
 
+    async fn merge_response_body(
```

Comment:
Interesting approach but I'm not sure that trying to handling requests and responses with the same function is worthwhile. Also the `Request` and `Response` are from the external `http` crate rather than a local implementation. Early return is a good point though.